sumo/src/normalize.rs

```diff
@@ -0,0 +1,149 @@
+                    junction_type: junction.junction_type,
+                    incoming_lanes: junction.incoming_lanes,
+                    internal_lanes: junction.internal_lanes,
+                    shape: junction.shape.unwrap(),
+                },
+            );
+        }
+
+        for edge in raw.edges {
+            if edge.function == raw::Function::Internal {
+                let mut lanes = Vec::new();
+                for lane in edge.lanes {
+                    lanes.push(InternalLane {
+                        id: lane.id,
+                        index: lane.index,
+                        speed: lane.speed,
+                        length: lane.length,
+                        center_line: lane.shape.ok(),
+                        allow: lane.allow,
+                    });
+                }
+                network
+                    .internal_edges
+                    .insert(edge.id.clone(), InternalEdge { id: edge.id, lanes });
+                continue;
+            }
+
+            let from = edge.from.unwrap();
+            let to = edge.to.unwrap();
+            let template = &types[edge.edge_type.as_ref().unwrap()];
+
+            let raw_center_line = match edge.shape {
+                Some(pl) => pl,
+                None => {
+                    PolyLine::must_new(vec![network.junctions[&from].pt, network.junctions[&to].pt])
+                }
+            };
+            // TODO I tried interpreting the docs and shifting left/right by 1x or 0.5x of the total
+            // road width, but the results don't look right.
+            let center_line = match edge.spread_type {
+                raw::SpreadType::Center => raw_center_line,
+                raw::SpreadType::Right => raw_center_line,
+                raw::SpreadType::RoadCenter => raw_center_line,
+            };
+
+            let mut lanes = Vec::new();
+            for lane in edge.lanes {
+                lanes.push(Lane {
+                    id: lane.id,
+                    index: lane.index,
+                    speed: lane.speed,
+                    length: lane.length,
+                    // https://sumo.dlr.de/docs/Simulation/SublaneModel.html
+                    width: lane.width.unwrap_or(Distance::meters(3.2)),
+                    center_line: lane.shape.unwrap(),
+                    allow: lane.allow,
+                });
+            }
+
+            network.normal_edges.insert(
+                edge.id.clone(),
+                Edge {
+                    id: edge.id,
+                    edge_type: edge.edge_type.unwrap(),
+                    name: edge.name,
+                    from,
+                    to,
+                    priority: edge.priority.unwrap_or_else(|| template.priority),
+                    lanes,
+                    center_line,
+                },
+            );
+        }
+
+        network.fix_coordinates();
+        network
+    }
+
+    fn fix_coordinates(&mut self) {
```

Comment:
something like:

/// normalize coordinates to map space

would be a good tldr;